docs/sources/mimir/migration-guide/migrating-from-cortex.md

```diff
@@ -228,76 +232,42 @@ You can update to the Grafana Mimir Helm chart from the Cortex Helm chart.
    b. Use `mimirtool` to update the configuration.
 
    ```bash
-   mimirtool config convert --yaml-file cortex.yaml
+   mimirtool config convert --yaml-file cortex.yaml --yaml-out mimir.yaml
+   ```
+
+   c. Clean up the generated YAML configuration.
+   You have to remove some fields that are generated by `mimirtool config convert` for the Helm chart installation, because the
+   mimir-distributed Helm chart has already set the default value for them. Use the following script to clean up
+   those fields:
+
+   ```bash
+   # require yq v4
+   yq -i 'del(.activity_tracker.filepath,.alertmanager.data_dir,.compactor.data_dir)' mimir.yaml
    ```
 
-   c. Place the updated configuration under the `mimir.config` key at the top level of your Helm values file.
+   d. At the top level of your custom Helm values file, put the updated configuration under the `mimir.structuredConfig` key.
 
-   > **Note:** The Grafana Mimir Helm chart expects the configuration as a string value.
-   > You can provide a literal block string with the `|` symbol.
+   > **Note:** The `mimir.structuredConfig` field, which is added in v3.0.0, allows you to override a specific
+   > configuration without needing to rewrite the whole block string literal, such as in `mimir.config`.
 
    In your Helm values file:
 
    ```yaml
    mimir:
-     config: |
-       <CONFIGURATION FILE CONTENTS>
+     structuredConfig: <CONFIGURATION FILE CONTENTS>
    ```
-
-   d. Merge the templated configuration from the `mimir-distributed` `values.yaml` file.
-   The Cortex Helm chart sets some additional configuration using flags.
-   The Grafana Mimir Helm chart sets that additional configuration in the configuration file.
-
-   - Set `frontend_worker.frontend_address` to `'{{ template "mimir.fullname" . }}-query-frontend-headless.{{ .Release.Namespace }}.svc:{{ include "mimir.serverGrpcListenPort" . }}'`.
-   - Set `ruler.alertmanager_url` to `'dnssrvnoa+http://_http-metrics._tcp.{{ template "mimir.fullname" . }}-alertmanager-headless.{{ .Release.Namespace }}.svc.cluster.local/alertmanager'`.
-   - If you want to use memberlist as the ring KV store, set `memberlist.join_members` to `['{{ include "mimir.fullname" . }}-gossip-ring']`.
-   - Append the caching configuration to `blocks_storage.bucket_store`.
-   - Set `ingester.ring.num_tokens` to the existing value you set in `ingester.lifecycler.num_tokens`.
-     This is especially important if you are using the default Cortex Helm chart values as it sets this to `512`.
```

Comment:
Are these instructions no longer relevant? I think they are because now they are set in the default value of `mimir.config`. But now this means that users should delete them from their structuredConfig, doesn't it?

Reply:
The users can delete them from their structured config but if it is also fine if they keep it because the final outcome is the same. Once template is expanded into string literal, it is difficult to track which expanded config refers to which template line. Hence I don't see it is worthed to track which line to delete. The only downside is now the structuredConfig will be a bit fatter than what it should be.

Reply:
my other consideration was that if we decide to change them in the future, it will be more difficult for some users and more people will have to take manual steps to go through it. i'm kind of ok to drop them. Otherwise you can include them in the `yq` command.

We can also change mimirtool config to have a `--helm` flag which does this removal automatically, but that will be more work.

Reply:
I think I just got to understand your original comment now. In that case, what I can do is add `frontend_worker.frontend_address` and `ruler.alertmanager_url` to the fields that yq will remove, therefore it will take the default values. 

I don't think we need to do anything on `memberlist.join_members` which something that user need to decide and `ingester.ring.num_tokens` which will be handled by mimir tool convert.